server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -226,6 +231,7 @@ final class DefaultSearchContext extends SearchContext {
         );
         queryBoost = request.indexBoost();
         this.lowLevelCancellation = lowLevelCancellation;
+        this.requestToAggReduceContextBuilder = requestToAggReduceContextBuilder;
```

Comment:
Why do need a builder here since we just need `InternalAggregation.ReduceContext` that we could build from the request?